repository_test.go

```diff
@@ -108,314 +112,316 @@ func createCommit(c *C, r *Repository) plumbing.Hash {
+	s.NoError(err)
 	// 6 refs total from github.com/git-fixtures/basic.git:
 	//  - HEAD
 	//  - refs/heads/master
 	//  - refs/heads/branch
 	//  - refs/pull/1/head
 	//  - refs/pull/2/head
 	//  - refs/pull/2/merge
-	c.Assert(count, Equals, 6)
+	s.Equal(6, count)
 
 	cfg, err := r.Config()
-	c.Assert(err, IsNil)
+	s.NoError(err)
 
-	c.Assert(cfg.Core.IsBare, Equals, true)
-	c.Assert(cfg.Remotes[DefaultRemoteName].Validate(), IsNil)
-	c.Assert(cfg.Remotes[DefaultRemoteName].Mirror, Equals, true)
+	s.True(cfg.Core.IsBare)
+	s.Nil(cfg.Remotes[DefaultRemoteName].Validate())
+	s.True(cfg.Remotes[DefaultRemoteName].Mirror)
 }
 
-func (s *RepositorySuite) TestCloneWithTags(c *C) {
+func (s *RepositorySuite) TestCloneWithTags() {
 	url := s.GetLocalRepositoryURL(
 		fixtures.ByURL("https://github.com/git-fixtures/tags.git").One(),
 	)
 
 	r, err := Clone(memory.NewStorage(), nil, &CloneOptions{URL: url, Tags: NoTags})
-	c.Assert(err, IsNil)
+	s.NoError(err)
 
 	remotes, err := r.Remotes()
-	c.Assert(err, IsNil)
-	c.Assert(remotes, HasLen, 1)
+	s.NoError(err)
+	s.Len(remotes, 1)
 
 	i, err := r.References()
-	c.Assert(err, IsNil)
+	s.NoError(err)
 
 	var count int
 	i.ForEach(func(r *plumbing.Reference) error { count++; return nil })
 
-	c.Assert(count, Equals, 3)
+	s.Equal(3, count)
 }
 
-func (s *RepositorySuite) TestCloneSparse(c *C) {
+func (s *RepositorySuite) TestCloneSparse() {
 	fs := memfs.New()
 	r, err := Clone(memory.NewStorage(), fs, &CloneOptions{
 		URL:        s.GetBasicLocalRepositoryURL(),
 		NoCheckout: true,
 	})
-	c.Assert(err, IsNil)
+	s.NoError(err)
 
 	w, err := r.Worktree()
-	c.Assert(err, IsNil)
+	s.NoError(err)
 
 	sparseCheckoutDirectories := []string{"go", "json", "php"}
-	c.Assert(w.Checkout(&CheckoutOptions{
+	s.NoError(w.Checkout(&CheckoutOptions{
 		Branch:                    "refs/heads/master",
 		SparseCheckoutDirectories: sparseCheckoutDirectories,
-	}), IsNil)
+	}))
 
 	fis, err := fs.ReadDir(".")
-	c.Assert(err, IsNil)
+	s.NoError(err)
 	for _, fi := range fis {
-		c.Assert(fi.IsDir(), Equals, true)
+		s.True(fi.IsDir())
 		var oneOfSparseCheckoutDirs bool
 
 		for _, sparseCheckoutDirectory := range sparseCheckoutDirectories {
 			if strings.HasPrefix(fi.Name(), sparseCheckoutDirectory) {
 				oneOfSparseCheckoutDirs = true
 			}
 		}
-		c.Assert(oneOfSparseCheckoutDirs, Equals, true)
+		s.True(oneOfSparseCheckoutDirs)
 	}
 }
 
-func (s *RepositorySuite) TestCreateRemoteAndRemote(c *C) {
+func (s *RepositorySuite) TestCreateRemoteAndRemote() {
 	r, _ := Init(memory.NewStorage(), nil)
 	remote, err := r.CreateRemote(&config.RemoteConfig{
 		Name: "foo",
 		URLs: []string{"http://foo/foo.git"},
 	})
 
-	c.Assert(err, IsNil)
-	c.Assert(remote.Config().Name, Equals, "foo")
+	s.NoError(err)
+	s.Equal("foo", remote.Config().Name)
 
 	alt, err := r.Remote("foo")
-	c.Assert(err, IsNil)
-	c.Assert(alt, Not(Equals), remote)
-	c.Assert(alt.Config().Name, Equals, "foo")
+	s.NoError(err)
+	s.NotEqual(remote, alt)
```

Comment:
The test is to ensure a new instance is returned, so we need to use `NotSame` instead of `NotEqual`:
```suggestion
	s.NotSame(remote, alt)
```

Reply:
@BeChris this fixes the issue you are experiencing with `TestCreateRemoteAndRemote`.